pkg/gpud-manager/systemd/systemd.go

```diff
@@ -36,3 +46,48 @@ FLAGS="--log-level=info --log-file=/var/log/gpud.log"
 `)
 	return err
 }
+
+func addLogFileFlagIfExists(file string) error {
+	if _, err := os.Stat(file); errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
+
+	readFile, err := os.OpenFile(file, os.O_RDONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer readFile.Close()
+
+	lines := make([]string, 0)
+	scanner := bufio.NewScanner(readFile)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+
+		// only edit the line that contains FLAGS=
+		if !strings.Contains(line, "FLAGS=") {
+			lines = append(lines, line)
+			continue
+		}
+
+		// FLAGS already contains --log-file flag
+		if strings.Contains(line, "--log-file=") {
+			lines = append(lines, line)
+			continue
+		}
+
+		// remove the trailing " character
+		line = strings.TrimSuffix(line, "\"")
+		line = fmt.Sprintf("%s --log-file=/var/log/gpud.log\"", line)
+
+		lines = append(lines, line)
+	}
```

Comment:
it's better to wrap the code as a function.

Reply:
done